test/integration/aggregator_test.go

```diff
@@ -0,0 +1,124 @@
+package integration
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiregistrationclientset "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+
+	"github.com/openshift/origin/pkg/cmd/server/admin"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	projectclientset "github.com/openshift/origin/pkg/project/generated/clientset"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+func TestAggregator(t *testing.T) {
```

Comment:
I did this upstream, but unless you create a service running an actual API server behind it, this test never catches the breaks you're interested in.  Also, this would be removed as soon as we merge the rebase and make the aggregator non-optional.